test: add eigenda v1 to v2 migration kurtosis test

samlaf opened the pull request:
Closes DAINT-388

Using the proxy v1.7.0-rc docker image
updated the test harness to add a client for the new proxy admin API which allows changing dispersal backend (v1 -> v2) added test to test this migration

samlaf committed:
test(kt-devnet): add v1->v2 migration test

Using the proxy v1.7.0-rc docker image
updated the test harness to add a client for the new proxy admin API which allows changing dispersal backend (v1 -> v2)
added test to test this migration

samlaf committed:
test(kt-devnet): fix test harness

proxyClient was not being initialized properly, causing a nil pointer error.

samlaf committed:
ci(kt-devnet): add v2 files to test matrix

samlaf committed:
docs(kt-devnet): fix comment on TestEigenDAV2Migration_Memstore test

litt3 commented on kurtosis-devnet/tests/eigenda/harness.go:
```diff
@@ -249,3 +261,90 @@ func (c *ProxyMemconfigClient) Failback(ctx context.Context) error {
+	proxyEndpoint string // e.g. http://localhost:3100
+}
+
+func (c *ProxyAdminAPIClient) GetDispersalBackend(ctx context.Context) (EigenDACertVersion, error) {
```
nit: unused `ctx` param (also for the `set` method below).

samlaf replied:
good catch: [c0d449d](https://github.com/Layr-Labs/optimism/pull/44/commits/c0d449d57e5a639053bfff6551092b34bf7b4586)

litt3 commented on kurtosis-devnet/tests/eigenda/v2_migration_test.go:
```diff
@@ -0,0 +1,61 @@
+// The batches inbox transactions are queried via geth's GraphQL API.
+//
+// Note: because this test modifies the proxy's state config, it should be run in isolation (sequentially).
+func TestEigenDAV2Migration_Memstore(t *testing.T) {
```
Based on the name, this is using `Memstore`. Any reason you're not using `Holesky`?

samlaf replied:
I didn't want holesky tests to start taking 40 mins (trying to stay around the 20 min mark). But I ugess this is only 4 mins, so I could run it on holesky as well.

samlaf replied:
added in [35a545b](https://github.com/Layr-Labs/optimism/pull/44/commits/35a545b265113838546dc303a968959c88f2bd1a), lets see how long CI takes with this.

samlaf committed:
test(kt-devnet): make GetDispersalBackend use passed ctx

samlaf committed:
test(kt-devnet): add finalization assert to migration test

We check that op-node is still finalizing after the migration, just like we do in the failover test.

samlaf committed:
test(kt-devnet): also test migration against holesky

samlaf commented on kurtosis-devnet/tests/eigenda/v2_migration_test.go:
```diff
@@ -0,0 +1,61 @@
+	_, err = geth.WaitForBlock(big.NewInt(int64(stage2ToBlockNum)), harness.Clients.GethL1Client)
+	require.NoError(t, err)
+
+	requireBatcherTxsToBeFromLayer(t, stage2FromBlockNum, stage2ToBlockNum, DALayerEigenDAV2, harness.Endpoints.GethL1Endpoint, harness.BatchInboxAddr)
```
[045c22a](https://github.com/Layr-Labs/optimism/pull/44/commits/045c22afa8120e22bc2982ea053e16670be606ba)

samlaf committed:
test(kt-devnet): fix wrong context bug

was passing old context in places, causing context timed out error

samlaf committed:
chore(kt-devnet): add max-blob-length flag to make proxy load faster

proxy takes ~1min to load and sometimes longer. add the --eigenda.v2.max-blob-length flag set to 1MiB to make loading faster (loads fewer SRS points at startup)

samlaf committed:
chore(kt-devnet): add justfile command to switch proxy backend

allows switching from v1 to v2 when doing manual testing locally

samlaf committed:
test(kt-devnet): add log line to show discarded commitments

in failover test

samlaf committed:
test(kt-devnet): skip migration test on holesky to make CI faster

Copy paste of code comment explaining reasoning:
	// The issue is that because of the holocene strict ordering rules,
	// after migrating from v1 to v2, the v2 certs, even though they finalize very quickly
	// and are returned to op-batcher, op-batcher cannot send them to the batch-inbox until
	// the v1 certs have been sent. So we need to wait at least 10-15 mins for the v1 certs to be returned
	// from proxy to op-batcher and submitted to batch-inbox.
	// starting kurtosis devnet takes ~10 mins and batcher logs tests take 15 mins,
	// so we are already at 25 mins. Trying to keep CI time under 30 mins,
	// so we skip this test for now. We could turn on if we had a nice inabox eigenda that would allow
	// for much faster v1 finalization.

samlaf committed:
test(kt-devnet): reenable v2migration test on holesky

Decided to live with increased CI time in order to be safe.

samlaf committed:
test(kt-devnet): fix migration test by increasing waitForBlock timer

samlaf committed:
chore(kt-devnet): bump eigenda.yaml proxy to v1.7.0

samlaf committed:
test(kt-devnet): add extra minute to WaitForBlock to prevent timeout

samlaf committed:
test(kt-devnet): increase go test timeout to 30/40mins

samlaf committed:
test(kt-devnet): bump v1 stage to 40 l1 blocks in migration test

samlaf committed:
test(kt-devnet): add extra minute to WaitForBlock to prevent timeout

samlaf committed:
test(kt-devnet): add -v flags to get more info in ci

samlaf committed:
test(kt-devnet): change max-channel-duration 150->10 for holesky tests

samlaf committed:
ci(kt-devnet): remove holesky-sequential-large-blobs v1 test

Kept having flukes. It was taking super long. Test getting throttled. Was just a ci nightmare, so opted to remove.

samlaf commented:
@litt3 finally ready for a review. Had to play around with the darn setup/tests a lot so forget all the choices I did lol.
Let me know if the changes since your last review make sense. Otherwise we can sync in person to discuss.

litt3 reviewed (approved):
Changes make sense to me
